cpp/src/arrow/dataset/scanner.cc

```diff
@@ -85,6 +85,8 @@ class ScannerRecordBatchReader : public RecordBatchReader {
     return Status::OK();
   }
 
+  Status Close() override { return Status::OK(); }
```

Comment:
We should execute exec plans until they finish.  So this should look something like...
```
Status Close() override {
  std::shared_ptr<RecordBatch> batch;
  RETURN_NOT_OK(ReadNext(&batch));
  while (batch != nullptr) {
    RETURN_NOT_OK(ReadNext(&batch));
  }
  return Status::OK();
}
```

Reply:
Sure.